Client/FluentClientExtensions.cs

```diff
@@ -21,8 +21,8 @@ public static class FluentClientExtensions
         public static bool Remove<TFilter>(this List<IHttpFilter> filters)
             where TFilter : IHttpFilter
         {
-            TFilter filter = filters.OfType<TFilter>().FirstOrDefault();
-            return filter != null && filters.Remove(filter);
+            IEnumerable<TFilter> tFilters = filters.OfType<TFilter>();
+            return tFilters.Any() && filters.Remove(tFilters.First());
```

Comment:
I was just trying to be as close to your original logic as possible while getting rid of the warning. If you want to remove all filters instead of just the fist one then you proposal makes sense.